src/game/enemies/yado.c

```diff
@@ -159,11 +159,11 @@ void Task_8055084(void)
             if (cheese->s.hitboxes[1].index != HITBOX_STATE_INACTIVE) {
                 if (HB_COLLISION(x, y, s->hitboxes[0], I(cheese->posX), I(cheese->posY), cheese->s.hitboxes[1])) {
                     if (IS_MULTI_PLAYER) {
-                        struct RoomEvent *roomEvent = CreateRoomEvent();
+                        struct RoomEvent_EnemyDestroy *roomEvent = (struct RoomEvent_EnemyDestroy *)CreateRoomEvent();
```

Comment:
You could make `CreateRoomEvent` a `void*` so that you don't have to cast like this every time? Not required though